### PR TITLE
added-tests-for-book-and-category

### PR DESCRIPTION
**Pull request by margosalii**

### CODE REVIEW

**Comment by margosalii** [comment]
Без имени

File: src/test/java/mate/academy/store/controller/BookControllerTest.java [hunk 1]
```diff
@@ -0,0 +1,254 @@
...
+import org.springframework.test.context.jdbc.Sql;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+import org.testcontainers.shaded.org.apache.commons.lang3.builder.EqualsBuilder;
+
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT)
+class BookControllerTest {
+    protected static MockMvc mockMvc;
```
Comment (MentorMateAcademy) [inline src/test/java/mate/academy/store/controller/BookControllerTest.java]: Why protected?

File: src/test/java/mate/academy/store/controller/BookControllerTest.java [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+    }
+
+    private static void generateBookDtos() {
+        bookDtos = new ArrayList<>();
+        bookDtos.add(new BookDto().setId(1L).setTitle("Harry Potter")
+                .setAuthor("J.K. Rowling").setIsbn("9780747532699")
+                .setPrice(BigDecimal.valueOf(11)).setCategoryIds(Collections.emptySet()));
+        bookDtos.add(new BookDto().setId(2L).setTitle("Treasure Island")
+                .setAuthor("Robert Louis Stevenson").setIsbn("9780141321004")
+                .setPrice(BigDecimal.valueOf(8)).setCategoryIds(Collections.emptySet()));
```
Comment (MentorMateAcademy) [inline src/test/java/mate/academy/store/controller/BookControllerTest.java]: write in stack form

File: src/test/java/mate/academy/store/controller/BookControllerTest.java [hunk 3]
```diff
@@ -0,0 +1,254 @@
...
+                get("/books").contentType(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andReturn();
+
+        BookDto[] actual = objectMapper.readValue(result
+                .getResponse()
+                .getContentAsByteArray(), BookDto[].class);
+
+        Assertions.assertEquals(2, actual.length);
+        Assertions.assertEquals(expected, Arrays.stream(actual).toList());
```
Comment (MentorMateAcademy) [inline src/test/java/mate/academy/store/controller/BookControllerTest.java]: you don`t need stream, Arrays.asList

File: src/test/java/mate/academy/store/controller/BookControllerTest.java [hunk 4]
```diff
@@ -0,0 +1,254 @@
...
+            executionPhase = Sql.ExecutionPhase.AFTER_TEST_METHOD
+    )
+    @DisplayName("Update book by ID")
+    void updateBook_validIdAndRequestDto_Ok() throws Exception {
+        CreateBookRequestDto requestDto = new CreateBookRequestDto();
+        requestDto.setAuthor("Shevchenko");
+        requestDto.setTitle("Kobzar");
+        requestDto.setDescription("Kobzar by Shenchenko");
+        requestDto.setPrice(BigDecimal.valueOf(16));
+        requestDto.setIsbn("9780141321002");
```
Comment (MentorMateAcademy) [inline src/test/java/mate/academy/store/controller/BookControllerTest.java]: if you use same strings just save it to constants

File: src/test/java/mate/academy/store/controller/CategoryControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,251 @@
...
+                .setName("Romantic").setDescription("Romantic books");
+
+        CategoryResponseDto expected = new CategoryResponseDto()
+                .setName(requestDto.getName())
+                .setDescription(requestDto.getDescription());
+
+        String jsonRequest = objectMapper.writeValueAsString(requestDto);
+
+        MvcResult result = mockMvc.perform(
+                put("/categories/1")
```
Comment (MentorMateAcademy) [inline src/test/java/mate/academy/store/controller/CategoryControllerTest.java:175]: constants

File: src/test/java/mate/academy/store/service/book/BookServiceTest.java [hunk 6]
```diff
@@ -0,0 +1,192 @@
...
+    private static Book book;
+    private static BookDto bookDto;
+    private static PageRequest pageRequest;
+    private static final Long ID = 1L;
+
+    @Mock
+    private BookRepository bookRepository;
+
+    @InjectMocks
+    private BookServiceImpl bookService;
```
Comment (MentorMateAcademy) [inline src/test/java/mate/academy/store/service/book/BookServiceTest.java]: inject place after mocks

**Review by MentorMateAcademy — CHANGES_REQUESTED** [review]

**Pull request merged**